### ISSUE TITLE
Add support to get env in custom router at server:start/server:run

### ISSUE THREAD

**Issue by lainosantos** [symfony/symfony#17077]
The commands server:start/server:run from FrameworkBundle have option to a custom router, however is not possible get the env options at the custom router.

**Comment by jakzal** [comment]
Currently a router corresponds to a specific environment. What are you trying to do?

**Comment by lainosantos** [comment]
It's ok. But when to run the commands server:start/server:run and set the option -r, anything request is routed to the new router file, and it's ok too. However in the new router file, don't have access to the --env argument, because any request, 
independently of environment, is routed to the same file, and that don't have how recover the env argument to decide the correct front controller file.

At the default command run, the code checks the --env and decides the specific environment router file.

I want know the --env argument inside the new router file, to decide the appropriate front controller.

**Comment by marfillaster** [comment]
This was supposed to be addressed by 696b978. However altered $_SERVER is not being passed to the frontend controller. I think `php -S` router has its own execution context.  

The proper fix should be this code in `app_dev.php`.
```
if (php_sapi_name() === 'cli-server') {
    $_SERVER = array_merge($_ENV, $_SERVER);
}
```

**Comment by nicolas-grekas** [comment]
Closing as things changed significantly in this area with Flex, so that this might not be relevant anymore.
Please resubmit if behavior with 3.4 is still not as you'd expect.
